### PR TITLE
WT-4995 Migrate Jenkins wiredtiger-test-wtperf job to Evergreen

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
Added a new task "test-wtperf" under Ubuntu 18.04 buildvariant

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -1586,6 +1586,27 @@ tasks:
...
+          script: |
+            set -o errexit
+            set -o verbose
+
+            mkdir WT_TEST
+            dir=../bench/wtperf/stress
+            for file in `ls $dir`
+            do
+              ./bench/wtperf/wtperf -O $dir/$file -o verbose=2
+              rm -rf WT_TEST/*
```
Comment (quchenhao) [inline test/evergreen.yml]: Can we change the code not overwriting WT_TEST?

File: test/evergreen.yml [hunk 2]
```diff
@@ -1586,6 +1586,27 @@ tasks:
           display_name: Coverage report
           remote_file: wiredtiger/${build_variant}/${revision}/coverage_report/coverage_report_${build_id}.html
 
+  - name: test-wtperf
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          posix_configure_flags: --enable-strict --enable-diagnostic
```
Comment (quchenhao) [inline test/evergreen.yml]: I think it's better to separate the compile task but in this case we only have one test. It's up to you.

File: test/evergreen.yml (hunk 1 above)
Comment (quchenhao) [reply test/evergreen.yml]: We should not remove the test output in evergreen.

**Comment by raviprakashgiri29** [comment]
Evergreen patch build

https://evergreen.mongodb.com/version/5dd356532fbabe0ba751883c

File: test/evergreen.yml (hunk 2 above)
Comment (lukech) [reply test/evergreen.yml]: I agree this should be separated, actually reusing the existing `compile` task is good enough (as . the configure options are a subset of that of `compile` task).

File: test/evergreen.yml [hunk 3]
```diff
@@ -1586,6 +1586,27 @@ tasks:
...
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/build_posix"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            mkdir WT_TEST
```
Comment (lukech) [inline test/evergreen.yml]: I know Python test suite will generate `WT_TEST` directory automatically, not sure about wtperf. We could save this line (though a comment would be helpful) if it's the case.

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml]: After rethinking this, I agree the test output could be useful for troubleshooting when the test fails. 
Even though Jenkins did not save the directory, we can simply do it by replacing this line with `cp WT_TEST WT_TEST_$file`.

File: test/evergreen.yml [hunk 4]
```diff
@@ -1586,6 +1586,27 @@ tasks:
           display_name: Coverage report
           remote_file: wiredtiger/${build_variant}/${revision}/coverage_report/coverage_report_${build_id}.html
 
+  - name: test-wtperf
```
Comment (lukech) [inline test/evergreen.yml]: Can we call it `wtperf-test` (to be consistent with existing Evergreen tasks)?

File: test/evergreen.yml (hunk 3 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Ok. Let me confirm and then remove the line if needed.

File: test/evergreen.yml [hunk 5]
```diff
@@ -1586,6 +1593,27 @@ tasks:
...
+      - func: "fetch artifacts"
+        vars:
+          dependent_task: compile-wtperf
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/build_posix"
+          script: |
+            set -o errexit
+            set -o verbose
+            # Python test suite generates WT_TEST directory
```
Comment (lukech) [inline test/evergreen.yml]: This comment is a bit confusing here as we're not running Python test suite in this task.

**Pull request merged**